python/lsst/pipe/tasks/photoCal.py

```diff
@@ -648,21 +670,42 @@ def run(self, exposure, sourceCat):
             frame = None
 
         res = self.loadAndMatch(exposure, sourceCat)
+
+        #from res.matches, reserve a fraction of the population and mark the sources as reserved
+        reserveList = []
```

Comment:
move `reserveList` inside the `if`, since we don't need it unless we're reserving things.